example/server.py

```diff
@@ -1,41 +1,40 @@
-from flask import Flask
-from flask_opentracing import FlaskTracer
-import lightstep.tracer
 import opentracing
-import urllib2
-
-app = Flask(__name__)
-
-# one-time tracer initialization code
-ls_tracer = lightstep.tracer.init_tracer(group_name="example server", access_token="{your_lightstep_token}")
-# This tracer traces all requests
-tracer = FlaskTracer(ls_tracer, True, app, ["url_rule"])
-
-@app.route("/simple")
-def simple_response():
-	'''
-	This request will be automatically traced.
-	'''
-	return "Hello, world!"
-
-@app.route("/childspan")
-def create_child_span():
-	'''
-	This request will also be automatically traced.
-	
-	This is a more complicated example of accessing the current 
-	request from within a handler and creating new spans manually.
-	'''
-	parent_span = tracer.get_span()
-	child_span = ls_tracer.start_span("inside create_child_span", child_of=parent_span)
-	ans = calculate_some_stuff()
-	child_span.finish()	
-	return str(ans)
-
-def calculate_some_stuff():
-	two = 1 + 1
-	return two
-
-
-
-
+import logging
+from jaeger_client import Config
+from flask_opentracing import FlaskTracer
+from opentracing.propagation import Format
+from flask import Flask, request
+from opentracing.ext import tags
+from os import environ
+
+JAEGER_HOST = environ.get('JAEGER_HOST')
+
+if __name__ == '__main__':
+        app = Flask(__name__)
+        log_level = logging.DEBUG
+        logging.getLogger('').handlers = []
+        logging.basicConfig(format='%(asctime)s %(message)s', level=log_level)
+        config = Config(config={'sampler': {'type': 'const', 'param': 1},
+                                'logging': True,
+                                'local_agent':
+                                {'reporting_host': JAEGER_HOST}},
+                        service_name="jaeger_opentracing_example")
+        ls_tracer = config.initialize_tracer()
+        tracer = FlaskTracer(ls_tracer)
+
+        @app.route('/log')
+        @tracer.trace()
+        def log():
+
+                span_ctx = ls_tracer.extract(Format.HTTP_HEADERS, request.headers)
```

Comment:
similar to client, the point of using `@tracer.trace()` is that the handlers are automatically instrumented, but you're doing it again manually